src/Microsoft.AspNetCore.Authorization/AuthorizationHandler.cs

```diff
@@ -6,38 +6,53 @@
 
 namespace Microsoft.AspNetCore.Authorization
 {
+    /// <summary>
+    /// Base class for authorization handlers that need to be called for a specific requirement type.
+    /// </summary>
+    /// <typeparam name="TRequirement">The type of the requirement to handle.</typeparam>
     public abstract class AuthorizationHandler<TRequirement> : IAuthorizationHandler
-        where TRequirement : IAuthorizationRequirement
+            where TRequirement : IAuthorizationRequirement
     {
-        public void Handle(AuthorizationHandlerContext context)
+        async Task IAuthorizationHandler.HandleAsync(AuthorizationHandlerContext context)
```

Comment:
My view on this is that this class is a helper class to provide a behavior (filtering on requirements) it doesn't make sense to rewrite it, or then the user would just have to implement the `IAuthorizationHandler` interface directly. On the opposite, making it public would let developers think then can override it instead of `Handle` or `HandleAsync`.